src/st/chi/dt_entity.c

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+/*
+ * File: dt_entity.c
+ * Overlay: CHI
+ * Description: DATA - Entities
+ */
+
+#include "chi.h"
+
+void EntityBreakableWithDebris(Entity*);
+void EntityExplosion(Entity*);
+void EntityPrizeDrop(Entity*);
+void EntityDamageDisplay(Entity*);
+void OVL_EXPORT(EntityRedDoor)(Entity*);
+void EntityIntenseExplosion(Entity*);
+void EntitySoulStealOrb(Entity*);
+void EntityRoomForeground(Entity*);
+void EntityStageNamePopup(Entity*);
+void EntityEquipItemDrop(Entity*);
+void EntityRelicOrb(Entity*);
+void EntityHeartDrop(Entity*);
+void EntityEnemyBlood(Entity*);
+void EntityMessageBox(Entity*);
+void EntityDummy(Entity*);
+void EntityBackgroundBlock(Entity*);
+void EntityUnkId12(Entity*);
+void EntityUnkId13(Entity*);
+void EntityExplosionVariants(Entity*);
+void EntityGreyPuff(Entity*);
+void EntityDemonSwitchWall(Entity*);
+void EntityDemonSwitch(Entity*);
+void EntityBreakableWall(Entity*);
+void EntityBreakableWallDebris(Entity*);
+void EntityBreakableDebris(Entity*);
+void EntityFallingStairs(Entity*);
+void EntityCerberusGateDebug(Entity*);
+void EntityFallingStep(Entity*);
+void EntityGremlin(Entity*);
+void EntityGremlinEffect(Entity*);
+void EntityGremlinFire(Entity*);
+void EntitySalemWitch(Entity*);
+void EntitySalemWitchGlow(Entity*);
+void EntitySalemWitchCurse(Entity*);
+void EntitySalemWitchTriboltLaunch(Entity*);
+void EntitySalemWitchTriboltProjectile(Entity*);
+void EntityThornweed(Entity*);
+void EntityCorpseweed(Entity*);
+void EntityCorpseweedProjectile(Entity*);
+void EntityVenusWeed(Entity*);
+void EntityVenusWeedFlower(Entity*);
+void EntityVenusWeedTendril(Entity*);
+void EntityVenusWeedDart(Entity*);
+void EntityVenusWeedSpike(Entity*);
+
+// D_8018055C
+PfnEntityUpdate OVL_EXPORT(EntityUpdates)[] = {
+    EntityBreakableWithDebris,
+    EntityExplosion,
+    EntityPrizeDrop,
+    EntityDamageDisplay,
+    OVL_EXPORT(EntityRedDoor),
+    EntityIntenseExplosion,
+    EntitySoulStealOrb,
+    EntityRoomForeground,
+    EntityStageNamePopup,
+    EntityEquipItemDrop,
+    EntityRelicOrb,
+    EntityHeartDrop,
+    EntityEnemyBlood,
+    EntityMessageBox,
+    EntityDummy,
+    EntityDummy,
+    EntityBackgroundBlock,
+    EntityUnkId12, // EntityCameraLock
+    EntityUnkId13, // EntityParticleTrail
```

Comment:
Just a suggestion, it might be good to add this as a comment to these two .h files and we can consider renaming them later on (I saw you added a comment about the params already)